src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobWorker.cs

```diff
@@ -65,8 +91,10 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                             foreach (ReindexJobWrapper job in jobs)
                             {
                                 _logger.LogTrace("Picked up reindex job: {jobId}.", job.JobRecord.Id);
-
-                                runningTasks.Add(_reindexJobTaskFactory().ExecuteAsync(job.JobRecord, job.ETag, cancellationToken));
+                                var source = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
>CancellationTokenSource.CreateLinkedTokenSource(cancellationToken) [](http://example.com/codeflow?start=45&length=66)

`ReindexJobWorker.ExecuteAsync` get `cancellationToken` from `ReindexJobWorkerBackgroundService` which is background service.

I assume if we stop service, that token become cancelled. By become cancelled it will trigger LinkedToken to be cancelled as well. 

Which will cancel current reindex job. Shouldn't we pause job instead of cancel it if we shutdown service?

Reply:
You are correct that we will not want to mark the job as cancelled.  We do want to shutdown any running tasks, which is the intent of linking the cancellation tokens.  I will double check that we not update the job record to Canceled unless that is the intent.